stackdriver-core/lib/stackdriver/core/configuration.rb

```diff
@@ -18,107 +18,99 @@ module Core
     ##
     # @private Helps organize configuration options for Stackdriver
     # instrumentation libraries. It's initialized with a nested list of
-    # predefined option keys, then only allows getting and setting these
+    # predefined category keys, then only allows getting and setting these
     # predefined options.
     #
     # @example
-    #   options = [:opt1, {category1: [:opt2]}, {category2: [:opt3]}]
-    #   config = Stackdriver::Core::Configuration.new options: options
+    #   nested_categories = [:cat1, {cat2: [:cat3]}]
+    #   config = Stackdriver::Core::Configuration.new nested_categories
     #
-    #   config.opt1 = true
-    #   config.category1.opt2 = false
+    #   config.opt1        #=> nil
+    #   config.opt1 = true #=> true
+    #   config.opt1        #=> true
     #
-    #   config.opt1           #=> true
-    #   config.category1.opt2 #=> false
-    #   config.category2.opt3 #=> nil
-    #   config.opt4           #=> RuntimeError: Unrecognized option: opt4
+    #   config.cat1           #=> <Stackdriver::Core::Configuration>
+    #   config.cat2.cat3   #=> <Stackdriver::Core::Configuration>
     #
-    class Configuration
+    class Configuration < Hash
```

Comment:
Try not to inherit from core classes. Use composition: use a hash member variable, and delegate methods.